File: pset1/cash.c

#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)

{

    float change;

//ask user how much change is needed

    do
    {
        printf("Please enter the ammount of change required:");
        change = get_float();
    }

    while (change < 0.0);

//calculates change due based upon input

    int input = round(change * 100);
    int quaters = input / 25;
    int dimes = (input % 25) / 10;
    int nickles = (input % 25 % 10) / 5;
    int pennies = (input % 5) / 1;

//calculates minimum coins needed and outputs to user

    int total = quaters + dimes + nickles + pennies;
    printf ("%d\n", total);

}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    print("Please enter the ammount of change required:", end = "")
    change = get_float()

    if change >= 0:
        break
    else:
        print("Provide a Positive Value")

value = round(change * 100)
quaters = value / 25
dimes = value % 25 / 10
nickles = value % 25 % 1 / 5
pennies = value % 5 / 1

total = quaters + dimes + nickles + pennies

print("{}".format(total))